Baer picker failing with fi_prep_cif_var error

**Description of the Problem**

I recently installed obspy on a new python install. Everything appeared to be working until I ran the tests for a package that made a call to the obspy.signal.trigger.pk_baer, which gives the following error:

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [5], in <module>
----> 1 p_pick, phase_info = pk_baer(trace.data, df, 20, 60, 7.0, 12.0, 100, 100)

File ~/Gits/obspy/obspy/signal/trigger.py:406, in pk_baer(reltrc, samp_int, tdownmax, tupevent, thr1, thr2, preset_len, p_dur, return_cf)
    402 # index in pk_mbaer.c starts with 1, 0 index is lost, length must be
    403 # one shorter
    404 args = (len(reltrc) - 1, C.byref(pptime), pfm, samp_int,
    405         tdownmax, tupevent, thr1, thr2, preset_len, p_dur, cf_p)
--> 406 errcode = clibsignal.ppick(reltrc, *args)
    407 if errcode != 0:
    408     raise MemoryError("Error in function ppick of mk_mbaer.c")

RuntimeError: ffi_prep_cif_var failed
```

It seems fairly likely to me that one of the libraries I have installed introduced a problem in this environment compared to my old environment. I'm just not sure how to begin figuring out the issue. I tried it in my old environment just to be sure and it works fine.

**Steps to Reproduce**
Running the tests yields the following failures:
FAILED obspy/signal/tests/test_trigger.py::TriggerTestCase::test_pk_baer
FAILED obspy/signal/tests/test_trigger.py::TriggerTestCase::test_pk_baer_cf
in addition to a third, unrelated error in TauPy

Additionally, going through the trigger/picker tutorial will also reproduce the problem (this is probably what the test code is doing, anyways):
```
from obspy.core import read
from obspy.signal.trigger import pk_baer
trace = read("https://examples.obspy.org/ev0_6.a01.gse2")[0]
df = trace.stats.sampling_rate
p_pick, phase_info = pk_baer(trace.data, df,20, 60, 7.0, 12.0, 100, 100)
```

**Versions**
ObsPy: 1.2.2.post0+547.g221c7ef4a9.obspy.master  (I tried both this and 1.2.2 as downloaded from conda-forge)
Python: 3.8.12
OS: Ubuntu 16

**Obspy Installation**
I tried installing obspy in two different ways and got the same result both ways: `conda install -c conda-forge obspy` (technically I used mamba, but that shouldn't matter) and from the latest master (`pip install -e {path_to_source}').

**Old Environment**
Obspy: 1.2.2 (installed from conda)
Python: 3.8.8
OS: Ubuntu 16

Thank you,
Shawn

--- a/obspy/signal/headers.py
+++ b/obspy/signal/headers.py
@@ -61,7 +61,7 @@ clibsignal.ppick.argtypes = [
     np.ctypeslib.ndpointer(dtype=np.float32, ndim=1,
                            flags='C_CONTIGUOUS'),
     C.c_int, C.POINTER(C.c_int), C.c_char_p, C.c_float, C.c_int, C.c_int,
-    C.c_float, C.c_float, C.c_int, C.c_int]
+    C.c_float, C.c_float, C.c_int, C.c_int, C.POINTER(C.c_float)]
 clibsignal.ppick.restype = C.c_int
 
 clibsignal.ar_picker.argtypes = [

--- a/obspy/signal/tests/test_trigger.py
+++ b/obspy/signal/tests/test_trigger.py
@@ -4,13 +4,11 @@ The obspy.signal.trigger test suite.
 """
 import gzip
 import os
-import sys
 import unittest
 import warnings
 from ctypes import ArgumentError
 
 import numpy as np
-import pytest
 from numpy.testing import assert_array_almost_equal, assert_array_equal
 
 from obspy import Stream, UTCDateTime, read
@@ -82,10 +80,6 @@ class TriggerTestCase(unittest.TestCase):
         self.assertRaises(ArgumentError, clibsignal.recstalta,
                           np.array([1], dtype=np.int32), charfct, ndat, 5, 10)
 
-    @pytest.mark.xfail(
-        os.environ.get('CI') == 'true'
-        and os.environ.get('RUNNER_OS') == 'Linux'
-        and '3.8.13' in sys.version, reason='see #2984')
     def test_pk_baer(self):
         """
         Test pk_baer against implementation for UNESCO short course
@@ -100,10 +94,6 @@ class TriggerTestCase(unittest.TestCase):
         self.assertEqual(nptime, 17545)
         self.assertEqual(pfm, 'IPU0')
 
-    @pytest.mark.xfail(
-        os.environ.get('CI') == 'true'
-        and os.environ.get('RUNNER_OS') == 'Linux'
-        and '3.8.13' in sys.version, reason='see #2984')
     def test_pk_baer_cf(self):
         """
         Test pk_baer against implementation for UNESCO short course

